torch/lib/THD/master_worker/master/Master.cpp

```diff
@@ -4,29 +4,36 @@
 #include "../worker/Worker.h"
 #include "../../process_group/General.hpp"
 
-
 namespace thd {
-
 namespace master {
 
 std::unique_ptr<MasterCommandChannel> masterCommandChannel;
 
 } // namespace master
-
 } // namespace thd
 
 using namespace thd;
+using namespace thd::master;
 
 bool THDMasterWorkerInit(THDChannelType channel_type) {
   if (!THDProcessGroupInit(channel_type)) return false;
 
   if (dataChannel->getRank() > 0) {
+    /* Worker initialization. Can fail at start but then goes into infinite loop
+     * in which waits for commands from master.
+     */
     THDWorkerMain();
+    return false;
   }
 
-  // TODO: initialize master
-  thd::master::masterCommandChannel.reset(new MasterCommandChannel());
+  /* Master initialization. We need to make sure that all connections which
```

Comment:
Oh, I missed that when I moved error_thread to `CommandChannel`. Thanks for catching that!